kong/plugins/proxy-cache/handler.lua

```diff
@@ -173,11 +176,25 @@ local function cacheable_response(conf, cc)
       return false
     end
 
+    local opts = { ignorecase_params = { "charset" } }
+    local t, subtype, params = parse_mime_type(content_type, opts)
     local content_match = false
     for i = 1, #conf.content_type do
-      if conf.content_type[i] == content_type then
-        content_match = true
-        break
+      local t1, subtype1, params1 = parse_mime_type(conf.content_type[i], opts)
```

Comment:
does `parse_mime_type` have a failure mode?

can we add a test that exercises that mode. It's not immediately obvious to me what would happen in this case :thinking:

Reply:
The util function described the return values of invalid input, see https://github.com/Kong/kong/pull/10030/files#diff-f3672f3ff7a49e3a18e9820b6146f8c41cf4f05e2c68d2ed99df3be71b1432d7R77

> -- @treturn nil|nil|nil Invalid mime-type

And thanks for pointing it out. I adjust it to handle the cases of parsing failure.